### PR TITLE
[Keyboard] add support for the kinT kinesis keyboard controller (kint2pp variant)

### PR DESCRIPTION
**Pull request by stapelberg**
## Description

This PR adds support for the kinT kinesis keyboard controller, which is a new revision of the older stapelberg controller.

See https://github.com/kinx-project/kint for the repository, https://michael.stapelberg.ch/posts/2020-07-09-kint-kinesis-keyboard-controller/ for the introduction blog post and https://youtu.be/I0kwQbnhlfk for the introduction stream, if you’re curious about more details.

I decided to send very small incremental commits in this PR for easier review: I branched keyboards/kinesis/stapelberg and then changed it to the new wiring. If you prefer to have one commit, feel free to squash and merge this PR.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kinesis/kint2pp/kint2pp.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+PC2 kAx Row: 0   P   SEM SLA RBR --  --
+PC3 kBx Row: MIN BSL APO RSH X1  --  --
+PC4 kCx Row: ESC F3  F6  F9  F12 PAU --
+PC5 kDx Row: F1  F4  F7  F10 PRT PRG --
+PC6 kEx Row: F2  F5  F8  F11 SLK KEY --
+*/
+
+
+
+#endif
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/kint2pp.h:104]:
```suggestion
```

File: keyboards/kinesis/kint2pp/kint2pp.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
+#ifndef KINESIS_KINT2PP_H
+#define KINESIS_KINT2PP_H
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/kint2pp.h:2]:
```suggestion
#pragma once
```

File: keyboards/kinesis/kint2pp/kint2pp.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+	// put your per-action keyboard code here
+	// runs for every action, just before processing by the firmware
+
+	return process_record_user(keycode, record);
+}
+
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/kint2pp.c:28]:
```suggestion

```

File: keyboards/kinesis/kint2pp/kint2pp.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
+#include "kint2pp.h"
+
+void matrix_init_kb(void) {
+	// put your keyboard start-up code here
+	// runs once when the firmware starts up
+    // * Set our LED pins as output
+    DDRC |= (1<<3); // Keypad LED: C3
+    DDRC |= (1<<4); // ScrLock LED: C4
+    DDRC |= (1<<5); // NumLock LED: C5
+    DDRC |= (1<<1); // CapsLock LED: C1
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/kint2pp.c:10]:
```suggestion
	led_init_ports();

```

File: keyboards/kinesis/kint2pp/kint2pp.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+	return process_record_user(keycode, record);
+}
+
+void led_init_ports() {
+  // * Set our LED pins as output
+  DDRC |= (1<<3); // Keypad LED: C3
+  DDRC |= (1<<4); // ScrLock LED: C4
+  DDRC |= (1<<5); // NumLock LED: C5
+  DDRC |= (1<<1); // CapsLock LED: C1
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/kint2pp.c:34]:
```suggestion
  setPinOutput(C3); // Keypad LED
  setPinOutput(C4); // ScrLock LED
  setPinOutput(C5); // NumLock LED
  setPinOutput(C1); // CapsLock LED
```

File: keyboards/kinesis/kint2pp/kint2pp.c [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  } else {
+      PORTC |= (1<<5);
+  }
+
+  if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+      PORTC &= ~(1<<1);
+  } else {
+      PORTC |= (1<<1);
+  }
+}
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/kint2pp.c:61]:
```suggestion
bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    if(res) {
        writePin(C3, !led_state.compose);
        writePin(C4, !led_state.scroll_lock);
        writePin(C5, !led_state.num_lock);
        writePin(C1, !led_state.caps_lock);
    }
    return res;
}
```

File: keyboards/kinesis/kint2pp/config.h [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+/* don't know if this should be defined at the board or top level. Assuming board 
+#define MOUSEKEY_DELAY          100
+#define MOUSEKEY_INTERVAL       20
+#define MOUSEKEY_MAX_SPEED      3
+#define MOUSEKEY_TIME_TO_MAX    10
+*/
+
+#define IGNORE_MOD_TAP_INTERRUPT
+
+#endif
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/config.h:72]:
```suggestion
```

File: keyboards/kinesis/kint2pp/config.h [hunk 8]
```diff
@@ -0,0 +1,72 @@
+#ifndef KINT2PP_CONFIG_H
+#define KINT2PP_CONFIG_H
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/config.h:2]:
```suggestion
#pragma once
```

File: keyboards/kinesis/kint2pp/config.h [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+#define MATRIX_COL_PINS \
+  {			\
+   E7, /* COL_0 */	\
+   F0, /* COL_1 */	\
+   F7, /* COL_2 */	\
+   B1, /* COL_3 */	\
+   B3, /* COL_4 */	\
+   B0, /* COL_5 */	\
+   D5 /* COL_6 */	\
+  }
```
Comment (fauxpark) [inline keyboards/kinesis/kint2pp/config.h:52]:
```suggestion
#define MATRIX_ROW_PINS { D7, E0, E1, C0, C6, F6, D4, D2, D3, D0, B7, D1, E6, B4, B2 }
#define MATRIX_COL_PINS { E7, F0, F7, B1, B3, B0, D5 }
```

Comment (stapelberg) [reply keyboards/kinesis/kint2pp/config.h:51]: Can you explain why you made the suggestion to remove the signal names next to the pins?

I thought including the signal names here would be helpful to people working with this hardware.

I certainly can make the change, it just seems like a step backwards to me, hence my question :)

Thanks!

**Comment by stapelberg** [comment]
Addressed all comments, and verified the current state does indeed compile and work on my keyboard.

Please take another look. Thank you!

File: keyboards/kinesis/kint2pp/config.h (hunk 9 above)
Comment (fauxpark) [reply keyboards/kinesis/kint2pp/config.h:51]: It's kinda pointless as you can easily figure out which column/row is which just by looking at the layout macros.

Comment (stapelberg) [reply keyboards/kinesis/kint2pp/config.h]: To me, it wasn’t obvious at all.

Anyway, done.

File: keyboards/kinesis/kint2pp/config.h [hunk 10]
```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "../config.h"
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/config.h:3]: This line _shouldn't_ be needed

```suggestion

#include "../config.h"
```

File: keyboards/kinesis/kint2pp/config.h [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* don't know if this should be defined at the board or top level. Assuming board 
+#define MOUSEKEY_DELAY          100
+#define MOUSEKEY_INTERVAL       20
+#define MOUSEKEY_MAX_SPEED      3
+#define MOUSEKEY_TIME_TO_MAX    10
+*/
+
```
Comment (zvecr) [inline keyboards/kinesis/kint2pp/config.h:41]: We would currently say keymap level on this sort of modification 
```suggestion
```

File: keyboards/kinesis/kint2pp/config.h (hunk 10 above)
Comment (stapelberg) [reply keyboards/kinesis/kint2pp/config.h]: Done

**Comment by stapelberg** [comment]
The PR Lint GitHub action is failing because of an unrelated issue with  `metamechs/timberwolf` AFAICT (from PR #9353 I suppose).

This should be ready to go. Please take another look. Thanks!

File: keyboards/kinesis/config.h [hunk 12]
```diff
@@ -136,5 +136,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #ifdef SUBPROJECT_stapelberg
     #include "stapelberg/config.h"
 #endif
+#ifdef SUBPROJECT_kint2pp
```
Comment (drashna) [inline keyboards/kinesis/config.h:139]: I don't believe that these are used anymore, actually.

Comment (noroadsleft) [reply keyboards/kinesis/config.h:139]: Yeah, Lines 133-141 aren't needed here at all, and deleting makes no difference to the build result.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**